### PR TITLE
feat: add the tabs generator

### PR DESCRIPTION
**Pull request by kedevked**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ng-primitives/ng-primitives/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What does this PR implement/fix?

- Add a generator for a tabs component

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by kedevked** [comment]
For this PR, I am generating 4 components. Let me know what you think @ashley-hunter

File: packages/ng-primitives/schematics/tabs/generator.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  generateFiles(tree, path.join(__dirname, `files/${panel}`), options.directory, {
+    ...options,
+    ...names(`${options.name}-${panel}`),
+    selector: `${options.prefix}-${fileName}-${panel}`,
+    styles: panelStyles,
+    template: `<ng-content />`,
+  });
+
+  generateFiles(tree, path.join(__dirname, `files/${panel}`), options.directory, {
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/tabs/generator.ts]: I think this should use `list` instead:

```suggestion
  generateFiles(tree, path.join(__dirname, `files/${list}`), options.directory, {
```

File: packages/ng-primitives/schematics/tabs/generator.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    ...options,
+    ...names(`${options.name}-${list}`),
+    selector: `${options.prefix}-${fileName}-${list}`,
+    styles: `:host { }`,
+    template: `<ng-content />`,
+  });
+
+  generateFiles(tree, path.join(__dirname, `files/${panel}`), options.directory, {
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/tabs/generator.ts]:
```suggestion
  generateFiles(tree, path.join(__dirname, `files/${button}`), options.directory, {
```

File: packages/ng-primitives/schematics/tabs/generator.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ...options,
+    ...names(`${options.name}-${button}`),
+    // selector: `${options.prefix}-${fileName}-${button}`,
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/tabs/generator.ts]: I think this can be removed

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Thank you! Looks good, just had a few small suggestions.

File: packages/ng-primitives/schematics/tabs/generator.ts (hunk 3 above)
Comment (kedevked) [reply packages/ng-primitives/schematics/tabs/generator.ts]: Do you mean, the button should not be generated ?

Comment (ashley-hunter) [reply packages/ng-primitives/schematics/tabs/generator.ts]: I was just referring to the commented out code 😊

Comment (kedevked) [reply packages/ng-primitives/schematics/tabs/generator.ts]: okay, I made the changes requested

**Review by ashley-hunter — APPROVED** [review]
Looks good, thank you

**Pull request merged**